ViAn/GUI/projectwidget.cpp

```diff
@@ -585,6 +617,44 @@ void ProjectWidget::context_menu(const QPoint &point) {
     menu.exec(mapToGlobal(point));
 }
 
+/**
+ * @brief ProjectWidget::tag_drawings
+ *  Creates a tag from all the drawings in the current video project
+ */
+void ProjectWidget::tag_drawings() {
+    VideoItem* vid_item = dynamic_cast<VideoItem*>(selectedItems().front());
+    BasicAnalysis* basic_tag = new DrawingTag();
+    basic_tag->m_name = "Drawings tag";
+    VideoProject* vid_proj = vid_item->get_video_project();
+    DrawingTag* tag = dynamic_cast<DrawingTag*>(basic_tag);
+
+    for (auto const& frame_overlay : vid_proj->get_overlay()->get_overlays()) {
+        if (frame_overlay.second.overlay.size() > 0) {
+            tag->add_frame(frame_overlay.first);
+        }
+    }
+    add_basic_analysis(vid_proj, basic_tag);
+}
+
+/**
+ * @brief ProjectWidget::update_tag_drawing
+ * Update the drawing tag so it matches with all the current drawings
+ */
+void ProjectWidget::update_tag_drawing() {
```

Comment:
Almost identical with `ProjectWidget::tag_drawings()`, could be merged into one function with a check if the tag should be updated?

Reply:
These aren't as identical as you say. The only thing identical is the loop but it might still be able to simplify.